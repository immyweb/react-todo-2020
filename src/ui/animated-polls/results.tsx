import * as React from 'react';

import { ChartHorizontalBar } from './horizontal-bar-chart';
import {
  APVote,
  APVoteImgBkgnd,
  APVoteHeadline,
  APCopy,
  APIntroLink,
  APIntroHashTag,
  APVoteContent,
  APVoteImgBkgndCover,
  APVoteImgBkgndCoverGradient,
  APResultsHeadline,
  APChartHolder,
} from './styles';

import { IResultsScreen, IVoteCount, ICard } from './types';

interface IProps extends IResultsScreen {
  options: ICard[];
}

export const processData = (cards: ICard[]): IVoteCount[] => {
  let totalVoteCount = 0;
  return cards.map(
    ({
      title,
      iconImg,
      voteCount: { voteCount, noVoteCount } = { voteCount: 0, noVoteCount: 0 },
    }) => {
      totalVoteCount += voteCount;
      const datum = {
        iconImg,
        voteCount,
        noVoteCount,
        player: title,
        get votePercentage(): number {
          return totalVoteCount ? Math.round((voteCount / totalVoteCount) * 100) : 0;
        },
      };
      return datum;
    },
  );
};

export const ResultsScreen: React.SFC<IProps> = ({
  bkgndImgMobile,
  bkgndImgDesktop,
  title,
  introTxt,
  options,
  ctaTxt,
  cta,
  ctaLink,
  ctaUrl,
  theme,
}) => {
  const graphData = processData(options);

  return (
    <APVote>
      <APVoteImgBkgnd imageMobile={bkgndImgMobile} imageDesktop={bkgndImgDesktop} />
      <APVoteImgBkgndCover />
      <APVoteImgBkgndCoverGradient />
      <APVoteContent>
        <APVoteHeadline>{title}</APVoteHeadline>
        <APCopy textAlign={'left'}>{introTxt}</APCopy>
        <APChartHolder>
          <ChartHorizontalBar data={graphData} theme={theme} />
        </APChartHolder>
        <APResultsHeadline>{ctaTxt}</APResultsHeadline>
        <APCopy>
          <APIntroLink href={ctaUrl}>
            {cta} <APIntroHashTag>{ctaLink}</APIntroHashTag>
          </APIntroLink>
        </APCopy>
      </APVoteContent>
    </APVote>
  );
};
