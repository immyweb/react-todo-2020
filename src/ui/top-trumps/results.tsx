import * as React from 'react';

import { IResultsScreen, IVoteCount, ICard } from './types';
import { ITheme } from '../../types/theme';
import { TTDefaultInner, TTResultsButtonLink, TTWrapper, TTResultsChartContainer } from './styles';
import { ChartHorizontalBar } from './horizontal-bar-chart';

interface IProps extends IResultsScreen {
  className: string;
  theme: ITheme;
}

export const processData = (cards: ICard[]): IVoteCount[] => {
  let totalVoteCount = 0;
  return cards.map(
    ({
      fName,
      sName,
      imageThumb,
      voteCount: { voteCount, noVoteCount } = { voteCount: 0, noVoteCount: 0 },
    }) => {
      totalVoteCount += voteCount;
      const datum = {
        imageThumb,
        voteCount,
        noVoteCount,
        player: `${fName} ${sName}`,
        get votePercentage(): number {
          return totalVoteCount ? Math.round((voteCount / totalVoteCount) * 100) : 0;
        },
      };
      return datum;
    },
  );
};

export const Results: React.SFC<IProps> = ({
  theme,
  cards,
  linkText,
  linkClick,
  imageDesktop,
  imageMobile,
}) => {
  const graphData = processData(cards);

  return (
    <TTDefaultInner imageMobile={imageMobile} imageDesktop={imageDesktop}>
      <TTWrapper>
        <TTResultsChartContainer>
          <ChartHorizontalBar data={graphData} theme={theme} />
        </TTResultsChartContainer>
        <TTResultsButtonLink
          onClick={linkClick}
          colour={theme.primary}
          hoverBackgroundColour={theme.secondary}
        >
          {linkText}
        </TTResultsButtonLink>
      </TTWrapper>
    </TTDefaultInner>
  );
};
