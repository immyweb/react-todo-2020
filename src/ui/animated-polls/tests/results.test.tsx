import * as React from 'react';
import { shallow } from 'enzyme';

import { ResultsScreen, processData } from '../results';

const resultsScreen = {
  bkgndImgMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg',
  bkgndImgDesktop: '',
  title: 'The Results',
  introTxt: 'Hereâ€™s what everyone else thought, did you agree with the others?',
  ctaTxt: 'Thanks for voting!',
  cta: 'Read more from:',
  ctaLink: '#MOVEMBER',
  ctaUrl: 'http://www.thesun.co.uk',
};

const cards = [
  {
    id: 'chevron',
    iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg',
    imageDesktop: '',
    title: 'The Chevron',
    copy: 'Looks a bit more Bruce Forsyth than Tom Selleck...',
    voteCount: {
      player: 'The Chevron',
      voteCount: 2,
      noVoteCount: 5,
    },
  },
  {
    id: 'handlebar',
    iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png',
    imageMobile: '',
    imageDesktop: '',
    title: 'The Handlebar',
    copy: 'Get a grip!',
    voteCount: {
      player: 'The Handlebar',
      voteCount: 5,
      noVoteCount: 2,
    },
  },
  {
    id: 'horsehoe',
    iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Horseshoe.png',
    imageMobile: '',
    imageDesktop: '',
    title: 'The Horseshoe',
    copy: 'Hulk Hogan would be proud',
    voteCount: {
      player: 'The Horseshoe',
      voteCount: 3,
      noVoteCount: 4,
    },
  },
  {
    id: 'pencil',
    iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png',
    imageMobile: '',
    imageDesktop: '',
    title: 'The Pencil',
    copy: 'Work it out with this',
    voteCount: {
      player: 'The Pencil',
      voteCount: 4,
      noVoteCount: 3,
    },
  },
];

const theme = {
  primary: '#000000',
  secondary: '#ffffff',
};

function shallowRender(props?: any) {
  return shallow(<ResultsScreen {...resultsScreen} options={cards} theme={theme} {...props} />);
}

describe('<ResultsScreen />', () => {
  describe('UI', () => {
    it('renders correctly', () => {
      const wrapper = shallowRender();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('processData', () => {
    it('returns array of vote results', () => {
      const data = processData([
        {
          id: 'chevron',
          iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png',
          imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg',
          imageDesktop: '',
          title: 'The Chevron',
          copy: 'Looks a bit more Bruce Forsyth than Tom Selleck...',
          voteCount: {
            player: 'The Chevron',
            voteCount: 2,
            noVoteCount: 5,
          },
        },
      ]);
      expect(data).toEqual([
        {
          player: 'The Chevron',
          voteCount: 2,
          noVoteCount: 5,
          votePercentage: 100,
          iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png',
        },
      ]);
    });

    it('returns zero votePercentage if no votes cast', () => {
      const data = processData([
        {
          id: 'chevron',
          iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png',
          imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg',
          imageDesktop: '',
          title: 'The Chevron',
          copy: 'Looks a bit more Bruce Forsyth than Tom Selleck...',
          voteCount: {
            player: 'The Chevron',
            voteCount: 0,
            noVoteCount: 0,
          },
        },
      ]);
      expect(data).toEqual([
        {
          player: 'The Chevron',
          voteCount: 0,
          noVoteCount: 0,
          votePercentage: 0,
          iconImg: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png',
        },
      ]);
    });
  });
});
