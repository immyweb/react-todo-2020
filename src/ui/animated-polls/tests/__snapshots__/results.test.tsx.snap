// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsScreen /> UI renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    bkgndImgDesktop=""
    bkgndImgMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg"
    cta="Read more from:"
    ctaLink="#MOVEMBER"
    ctaTxt="Thanks for voting!"
    ctaUrl="http://www.thesun.co.uk"
    introTxt="Here’s what everyone else thought, did you agree with the others?"
    options={
      Array [
        Object {
          "copy": "Looks a bit more Bruce Forsyth than Tom Selleck...",
          "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png",
          "id": "chevron",
          "imageDesktop": "",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg",
          "title": "The Chevron",
          "voteCount": Object {
            "noVoteCount": 5,
            "player": "The Chevron",
            "voteCount": 2,
          },
        },
        Object {
          "copy": "Get a grip!",
          "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png",
          "id": "handlebar",
          "imageDesktop": "",
          "imageMobile": "",
          "title": "The Handlebar",
          "voteCount": Object {
            "noVoteCount": 2,
            "player": "The Handlebar",
            "voteCount": 5,
          },
        },
        Object {
          "copy": "Hulk Hogan would be proud",
          "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Horseshoe.png",
          "id": "horsehoe",
          "imageDesktop": "",
          "imageMobile": "",
          "title": "The Horseshoe",
          "voteCount": Object {
            "noVoteCount": 4,
            "player": "The Horseshoe",
            "voteCount": 3,
          },
        },
        Object {
          "copy": "Work it out with this",
          "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png",
          "id": "pencil",
          "imageDesktop": "",
          "imageMobile": "",
          "title": "The Pencil",
          "voteCount": Object {
            "noVoteCount": 3,
            "player": "The Pencil",
            "voteCount": 4,
          },
        },
      ]
    }
    theme={
      Object {
        "primary": "#000000",
        "secondary": "#ffffff",
      }
    }
    title="The Results"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(div)
          imageDesktop=""
          imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg"
        />,
        <glamorous(div) />,
        <glamorous(div) />,
        <glamorous(div)>
          <glamorous(glamorous(glamorous(h2)))>
            The Results
          </glamorous(glamorous(glamorous(h2)))>
          <glamorous(glamorous(glamorous.P))
            textAlign="left"
          >
            Here’s what everyone else thought, did you agree with the others?
          </glamorous(glamorous(glamorous.P))>
          <glamorous(div)>
            <ChartHorizontalBar
              data={
                Array [
                  Object {
                    "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png",
                    "noVoteCount": 5,
                    "player": "The Chevron",
                    "voteCount": 2,
                    "votePercentage": 14,
                  },
                  Object {
                    "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png",
                    "noVoteCount": 2,
                    "player": "The Handlebar",
                    "voteCount": 5,
                    "votePercentage": 36,
                  },
                  Object {
                    "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Horseshoe.png",
                    "noVoteCount": 4,
                    "player": "The Horseshoe",
                    "voteCount": 3,
                    "votePercentage": 21,
                  },
                  Object {
                    "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png",
                    "noVoteCount": 3,
                    "player": "The Pencil",
                    "voteCount": 4,
                    "votePercentage": 29,
                  },
                ]
              }
              theme={
                Object {
                  "primary": "#000000",
                  "secondary": "#ffffff",
                }
              }
            />
          </glamorous(div)>
          <glamorous(glamorous(glamorous(glamorous(h2))))>
            Thanks for voting!
          </glamorous(glamorous(glamorous(glamorous(h2))))>
          <glamorous(glamorous(glamorous.P))>
            <glamorous(a)
              href="http://www.thesun.co.uk"
            >
              Read more from:
               
              <glamorous(span)>
                #MOVEMBER
              </glamorous(span)>
            </glamorous(a)>
          </glamorous(glamorous(glamorous.P))>
        </glamorous(div)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "imageDesktop": "",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(glamorous(glamorous(h2)))>
              The Results
            </glamorous(glamorous(glamorous(h2)))>,
            <glamorous(glamorous(glamorous.P))
              textAlign="left"
            >
              Here’s what everyone else thought, did you agree with the others?
            </glamorous(glamorous(glamorous.P))>,
            <glamorous(div)>
              <ChartHorizontalBar
                data={
                  Array [
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png",
                      "noVoteCount": 5,
                      "player": "The Chevron",
                      "voteCount": 2,
                      "votePercentage": 14,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png",
                      "noVoteCount": 2,
                      "player": "The Handlebar",
                      "voteCount": 5,
                      "votePercentage": 36,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Horseshoe.png",
                      "noVoteCount": 4,
                      "player": "The Horseshoe",
                      "voteCount": 3,
                      "votePercentage": 21,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png",
                      "noVoteCount": 3,
                      "player": "The Pencil",
                      "voteCount": 4,
                      "votePercentage": 29,
                    },
                  ]
                }
                theme={
                  Object {
                    "primary": "#000000",
                    "secondary": "#ffffff",
                  }
                }
              />
            </glamorous(div)>,
            <glamorous(glamorous(glamorous(glamorous(h2))))>
              Thanks for voting!
            </glamorous(glamorous(glamorous(glamorous(h2))))>,
            <glamorous(glamorous(glamorous.P))>
              <glamorous(a)
                href="http://www.thesun.co.uk"
              >
                Read more from:
                 
                <glamorous(span)>
                  #MOVEMBER
                </glamorous(span)>
              </glamorous(a)>
            </glamorous(glamorous(glamorous.P))>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "The Results",
            },
            "ref": null,
            "rendered": "The Results",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Here’s what everyone else thought, did you agree with the others?",
              "textAlign": "left",
            },
            "ref": null,
            "rendered": "Here’s what everyone else thought, did you agree with the others?",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ChartHorizontalBar
                data={
                  Array [
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png",
                      "noVoteCount": 5,
                      "player": "The Chevron",
                      "voteCount": 2,
                      "votePercentage": 14,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png",
                      "noVoteCount": 2,
                      "player": "The Handlebar",
                      "voteCount": 5,
                      "votePercentage": 36,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Horseshoe.png",
                      "noVoteCount": 4,
                      "player": "The Horseshoe",
                      "voteCount": 3,
                      "votePercentage": 21,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png",
                      "noVoteCount": 3,
                      "player": "The Pencil",
                      "voteCount": 4,
                      "votePercentage": 29,
                    },
                  ]
                }
                theme={
                  Object {
                    "primary": "#000000",
                    "secondary": "#ffffff",
                  }
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "data": Array [
                  Object {
                    "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png",
                    "noVoteCount": 5,
                    "player": "The Chevron",
                    "voteCount": 2,
                    "votePercentage": 14,
                  },
                  Object {
                    "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png",
                    "noVoteCount": 2,
                    "player": "The Handlebar",
                    "voteCount": 5,
                    "votePercentage": 36,
                  },
                  Object {
                    "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Horseshoe.png",
                    "noVoteCount": 4,
                    "player": "The Horseshoe",
                    "voteCount": 3,
                    "votePercentage": 21,
                  },
                  Object {
                    "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png",
                    "noVoteCount": 3,
                    "player": "The Pencil",
                    "voteCount": 4,
                    "votePercentage": 29,
                  },
                ],
                "theme": Object {
                  "primary": "#000000",
                  "secondary": "#ffffff",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Thanks for voting!",
            },
            "ref": null,
            "rendered": "Thanks for voting!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <glamorous(a)
                href="http://www.thesun.co.uk"
              >
                Read more from:
                 
                <glamorous(span)>
                  #MOVEMBER
                </glamorous(span)>
              </glamorous(a)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Read more from:",
                  " ",
                  <glamorous(span)>
                    #MOVEMBER
                  </glamorous(span)>,
                ],
                "href": "http://www.thesun.co.uk",
              },
              "ref": null,
              "rendered": Array [
                "Read more from:",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "#MOVEMBER",
                  },
                  "ref": null,
                  "rendered": "#MOVEMBER",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(div)
            imageDesktop=""
            imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg"
          />,
          <glamorous(div) />,
          <glamorous(div) />,
          <glamorous(div)>
            <glamorous(glamorous(glamorous(h2)))>
              The Results
            </glamorous(glamorous(glamorous(h2)))>
            <glamorous(glamorous(glamorous.P))
              textAlign="left"
            >
              Here’s what everyone else thought, did you agree with the others?
            </glamorous(glamorous(glamorous.P))>
            <glamorous(div)>
              <ChartHorizontalBar
                data={
                  Array [
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png",
                      "noVoteCount": 5,
                      "player": "The Chevron",
                      "voteCount": 2,
                      "votePercentage": 14,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png",
                      "noVoteCount": 2,
                      "player": "The Handlebar",
                      "voteCount": 5,
                      "votePercentage": 36,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Horseshoe.png",
                      "noVoteCount": 4,
                      "player": "The Horseshoe",
                      "voteCount": 3,
                      "votePercentage": 21,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png",
                      "noVoteCount": 3,
                      "player": "The Pencil",
                      "voteCount": 4,
                      "votePercentage": 29,
                    },
                  ]
                }
                theme={
                  Object {
                    "primary": "#000000",
                    "secondary": "#ffffff",
                  }
                }
              />
            </glamorous(div)>
            <glamorous(glamorous(glamorous(glamorous(h2))))>
              Thanks for voting!
            </glamorous(glamorous(glamorous(glamorous(h2))))>
            <glamorous(glamorous(glamorous.P))>
              <glamorous(a)
                href="http://www.thesun.co.uk"
              >
                Read more from:
                 
                <glamorous(span)>
                  #MOVEMBER
                </glamorous(span)>
              </glamorous(a)>
            </glamorous(glamorous(glamorous.P))>
          </glamorous(div)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "imageDesktop": "",
            "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(glamorous(glamorous(h2)))>
                The Results
              </glamorous(glamorous(glamorous(h2)))>,
              <glamorous(glamorous(glamorous.P))
                textAlign="left"
              >
                Here’s what everyone else thought, did you agree with the others?
              </glamorous(glamorous(glamorous.P))>,
              <glamorous(div)>
                <ChartHorizontalBar
                  data={
                    Array [
                      Object {
                        "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png",
                        "noVoteCount": 5,
                        "player": "The Chevron",
                        "voteCount": 2,
                        "votePercentage": 14,
                      },
                      Object {
                        "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png",
                        "noVoteCount": 2,
                        "player": "The Handlebar",
                        "voteCount": 5,
                        "votePercentage": 36,
                      },
                      Object {
                        "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Horseshoe.png",
                        "noVoteCount": 4,
                        "player": "The Horseshoe",
                        "voteCount": 3,
                        "votePercentage": 21,
                      },
                      Object {
                        "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png",
                        "noVoteCount": 3,
                        "player": "The Pencil",
                        "voteCount": 4,
                        "votePercentage": 29,
                      },
                    ]
                  }
                  theme={
                    Object {
                      "primary": "#000000",
                      "secondary": "#ffffff",
                    }
                  }
                />
              </glamorous(div)>,
              <glamorous(glamorous(glamorous(glamorous(h2))))>
                Thanks for voting!
              </glamorous(glamorous(glamorous(glamorous(h2))))>,
              <glamorous(glamorous(glamorous.P))>
                <glamorous(a)
                  href="http://www.thesun.co.uk"
                >
                  Read more from:
                   
                  <glamorous(span)>
                    #MOVEMBER
                  </glamorous(span)>
                </glamorous(a)>
              </glamorous(glamorous(glamorous.P))>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "The Results",
              },
              "ref": null,
              "rendered": "The Results",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Here’s what everyone else thought, did you agree with the others?",
                "textAlign": "left",
              },
              "ref": null,
              "rendered": "Here’s what everyone else thought, did you agree with the others?",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ChartHorizontalBar
                  data={
                    Array [
                      Object {
                        "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png",
                        "noVoteCount": 5,
                        "player": "The Chevron",
                        "voteCount": 2,
                        "votePercentage": 14,
                      },
                      Object {
                        "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png",
                        "noVoteCount": 2,
                        "player": "The Handlebar",
                        "voteCount": 5,
                        "votePercentage": 36,
                      },
                      Object {
                        "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Horseshoe.png",
                        "noVoteCount": 4,
                        "player": "The Horseshoe",
                        "voteCount": 3,
                        "votePercentage": 21,
                      },
                      Object {
                        "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png",
                        "noVoteCount": 3,
                        "player": "The Pencil",
                        "voteCount": 4,
                        "votePercentage": 29,
                      },
                    ]
                  }
                  theme={
                    Object {
                      "primary": "#000000",
                      "secondary": "#ffffff",
                    }
                  }
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "data": Array [
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Chevron.png",
                      "noVoteCount": 5,
                      "player": "The Chevron",
                      "voteCount": 2,
                      "votePercentage": 14,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Handlebar.png",
                      "noVoteCount": 2,
                      "player": "The Handlebar",
                      "voteCount": 5,
                      "votePercentage": 36,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Horseshoe.png",
                      "noVoteCount": 4,
                      "player": "The Horseshoe",
                      "voteCount": 3,
                      "votePercentage": 21,
                    },
                    Object {
                      "iconImg": "https://www.thesun.co.uk/wp-content/uploads/2018/10/Pencil.png",
                      "noVoteCount": 3,
                      "player": "The Pencil",
                      "voteCount": 4,
                      "votePercentage": 29,
                    },
                  ],
                  "theme": Object {
                    "primary": "#000000",
                    "secondary": "#ffffff",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Thanks for voting!",
              },
              "ref": null,
              "rendered": "Thanks for voting!",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <glamorous(a)
                  href="http://www.thesun.co.uk"
                >
                  Read more from:
                   
                  <glamorous(span)>
                    #MOVEMBER
                  </glamorous(span)>
                </glamorous(a)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "Read more from:",
                    " ",
                    <glamorous(span)>
                      #MOVEMBER
                    </glamorous(span)>,
                  ],
                  "href": "http://www.thesun.co.uk",
                },
                "ref": null,
                "rendered": Array [
                  "Read more from:",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "#MOVEMBER",
                    },
                    "ref": null,
                    "rendered": "#MOVEMBER",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
